### ISSUE TITLE
Does this library works good in recognizing faces of black women ? 

### ISSUE THREAD

**Issue by MLDSBigGuy** [ageitgey/face_recognition#362]
In this video https://www.youtube.com/watch?time_continue=6&v=TWWsW1w-BVo, speaker describes about how different libraries misclassified here and not able to detect her face. 

If this is because of the training samples dlib gave to train the model, how can we retrain on top of the already dlib trained models with our failed samples ?

Thank you,

**Comment by ageitgey** [comment]
The problem she talks about is a real and important problem. Most models are trained on whatever datasets are most easily available (for example, pictures of celebrities who have lots of images online). This means that the models can do worse for people not well represented in the training data. The trainined models will reflect whatever biases exist in the training data. For example, this module tends to have lower accuracy with people from Asia.

The way to fix the problem is to improve the training data to have a wider range of people. But that's surprisingly difficult because getting access to millions of labeled pictures of people's faces is not easy if you aren't Google or Facebook.

Face detection accuracy can be improved with relatively few images, but face recognition requires millions of images to train the model. Just adding in a few new samples won't help anything. You need to add hundreds of thousands of new samples.

As such, I don't know the answer to this problem yet. To make this program better, I wish there was an open source set of face images where people could contribute images for the benefit of everyone. But at the same time, there are obvious problems with building a public dataset like that with personal data.

To answer your specific question, it's not really practical for you to try to retain the face recognition model yourself. To do that, you'd need a few million pictures of faces separated by identity and a relatively beefy GPU. And even if someone else already has a dataset of millions of pictures like that for you to start with, those datasets aren't easy to share for various legal reasons (copyright, personal privacy, etc).

**Comment by MLDSBigGuy** [comment]
Thank you for the elaborate answer 👍

**Comment by kaisark** [comment]
Part of the problem is bias in the training set (algorithm/software), but part of facial detection and recognition limitations are also technical (lighting/sensor data - hardware). For now, the general rule is who ever has the most data (or defend-able data) wins...

Anyway, I pulled down her Ted photo (https://www.ted.com/speakers/joy_buolamwini) and ran the find_faces_in_picture_cnn.py script and it did locate her face successfully (results below):

https://www.media.mit.edu/projects/gender-shades/overview
http://gendershades.org/overview.html
"Researcher Joy Buolamwini initiated a systematic investigation after testing her TED speaker photo on facial analysis technology from leading companies. Some companies did not detect her face. Others labeled her face as male. After analyzing results on 1270 uniques faces, the Gender Shades authors uncovered severe gender and skin-type bias in gender classification."

![ss1](https://user-images.githubusercontent.com/3580312/36526876-d25b5e18-177d-11e8-891f-465d37316166.png)

![joybuolamwini_2016x1](https://user-images.githubusercontent.com/3580312/36526906-e8017414-177d-11e8-80a8-e3084cb214ea.jpg)
